src/spikeinterface/core/loading.py

```diff
@@ -18,110 +29,274 @@ def load(file_or_folder_or_dict, base_folder=None) -> BaseExtractor:
+        return analyzer
+
+    elif object_type in ("Recording", "Sorting", "Recording|Sorting"):
+        if (folder / f"si_folder.json").is_file():
+            # In later versions (0.94<v<0.102) we use the si_folder.json file
+            # This should be Recording | Sorting
+            si_file = folder / f"si_folder.json"
+        else:
+            # For backward compatibility (v<=0.94) we check for the cached.json/pkl/pickle files
+            for dump_ext in ("json", "pkl", "pickle"):
+                f = folder / f"cached.{dump_ext}"
+                if f.is_file():
+                    si_file = f
+        return BaseExtractor.load(si_file, base_folder=folder)
+
+
+def _guess_object_from_zarr(zarr_folder):
+    # here it can be a zarr folder for Recording|Sorting|SortingAnalyzer|Template
+    from .zarrextractors import super_zarr_open
+
+    zarr_root = super_zarr_open(zarr_folder, mode="r")
+
+    # can be SortingAnalyzer
+    spikeinterface_info = zarr_root.attrs.get("spikeinterface_info")
+    if spikeinterface_info is not None:
+        return _guess_object_from_dict(spikeinterface_info)
+
+    # here it is the old fashion and a bit ambiguous
+    if "templates_array" in zarr_root.keys():
+        return "Templates"
+    elif "channel_ids" in zarr_root.keys() and "unit_ids" not in zarr_root.keys():
+        return "Recording"
+    elif "unit_ids" in zarr_root.keys() and "channel_ids" not in zarr_root.keys():
+        return "Sorting"
+
+
+def _load_object_from_zarr(folder_or_url, object_type, **kwargs):
+    if object_type == "SortingAnalyzer":
+        from .sortinganalyzer import load_sorting_analyzer
+
+        backend_options = kwargs.get("backend_options", None)
+        load_extensions = kwargs.get("load_extensions", True)
+        analyzer = load_sorting_analyzer(
+            folder_or_url, backend_options=backend_options, load_extensions=load_extensions
+        )
+        return analyzer
+    elif object_type == "Templates":
+        from .template import Templates
+
+        templates = Templates.from_zarr(folder_or_url)
+        return templates
+    elif object_type == "Recording":
+        from .zarrextractors import read_zarr_recording
+
+        storage_options = kwargs.get("storage_options", None)
+        recording = read_zarr_recording(folder_or_url, storage_options=storage_options)
+        return recording
+    elif object_type == "Sorting":
+        from .zarrextractors import read_zarr_sorting
+
+        storage_options = kwargs.get("storage_options", None)
+        sorting = read_zarr_sorting(folder_or_url, storage_options=storage_options)
+        return sorting
+    elif object_type == "Recording|Sorting":
+        # This case shoudl deprecated soon because the read_zarr is ultra ambiguous
+        # just testing if the zarr contains unit_ids or channel_ids but many object also contains it (see template)!!!!
+        from .zarrextractors import read_zarr
```

Comment:
@alejoe91 : any idea how to avoid using this read_zarr stuff ?
If think that read_zarr should deprecated because this is too ambiguous.